components/webvr/webvr_thread.rs

```diff
@@ -0,0 +1,349 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use ipc_channel::ipc;
+use ipc_channel::ipc::{IpcReceiver, IpcSender};
+use msg::constellation_msg::PipelineId;
+use script_traits::{ConstellationMsg, WebVREventMsg};
+use std::{thread, time};
+use std::collections::{HashMap, HashSet};
+use std::sync::mpsc;
+use std::sync::mpsc::{Receiver, Sender};
+use webrender_traits;
+use webvr_traits::{WebVRMsg, WebVRResult};
+use webvr_traits::webvr::*;
+
+/// Defines the polling interval time in ms for VR Events such as VRDisplay connected, disconnected, etc.
+const EVENT_POLLING_INTERVAL: u64 = 500;
+
+/// WebVRThread owns native VRDisplays, handles their life cycle inside Servo and
+/// acts a doorman for untrusted VR requests from DOM Objects.
+/// It waits for VR Commands from DOM objects and handles them in its trusted thread.
```

Comment:
Could you add some more info on the design here? There are some long-running and short-lived threads plus communication back with the constellation and the lifetimes and expectations for some of the parts was not totally clear without a pretty deep reading. e.g., which threads have a lifetime tied to a pipeline vs. a single GetDisplays request vs. etc.